docs/proposal/20211207-delegated-storage.md

```diff
@@ -0,0 +1,554 @@
+                  "type": "string"
+                },
+                "namespace": {
+                  "$id": "#/properties/additionalParameters/properties/namespace",
+                  "type": "string"
+                }
+              },
+              "additionalProperties": false
+            }
+          }
+        },
+        "additionalProperties": false
+      }
+    ```
+
+    **TODO:** How we can enforce such convention? Type composition?
+
+1. To install new Storage Backend, Cluster Admin has two options:
+
+    - use Capact Actions (e.g. `cap.interface.capactio.capact.hub.storage.helm-release.install`).
+    - Register a storage backend by creating such TypeInstance.
+
+    Regardless the option, at the end there is one TypeInstance produced:
+      
+      ```yaml
+      id: 3ef2e4ac-9070-4093-a3ce-142139fd4a16
+      metadata:
+        name: helm-storage
+      typeRef:
+        path: cap.type.helm.storage
+        revision: 0.1.0
+      latestResourceVersion:
+        metadata:
+          attributes:
+          - path: cap.core.attribute.hub.storage.backend # related to GrapHQL implementation
+            revision: 0.1.0
+        value:
+          url: "helm-release.default:50051"
+          additionalParametersSchema: {
+            "$id": "#/properties/additionalParametersSchema",
+            "type": "object",
+            "required": [
+              "name",
+              "namespace"
+            ],
+            "properties": {
+              "name": {
+                "$id": "#/properties/additionalParametersSchema/properties/name",
+                "type": "string"
+              },
+              "namespace": {
+                "$id": "#/properties/additionalParametersSchema/properties/namespace",
+                "type": "string"
+              }
+            },
+            "additionalProperties": false
+          }
+      backend: capact-redis # immutable - contains TypeInstance ID or unique alias
+        # if not provided, fallback to default one 
+      ```
+
+1. In fresh Capact installation, there is one TypeInstance already preregistered:
+
+    **TODO:** Would Redis be really the default storage? Probably PostgreSQL will be better, as we treat it as a "builtin" Backend as well (to store Attributes, relations, etc.)
+
+    ```yaml
+    id: a36ed738-dfe7-45ec-acd1-8e44e8db893b
+    name: capact-redis
+    typeRef:
+        path: cap.core.type.hub.storage.redis
+        revision: 0.1.0
+    latestResourceVersion:
+      metadata:
+        attributes:
+        - path: cap.core.attribute.default # if more such Typeinstances with default Attribute, select first one
+          revision: 0.1.0 
+        - path: cap.core.attribute.hub.storage.backend
+          revision: 0.1.0
+      value:
+        url: "storagebackend-handlers.capact-system:50051"
+        additionalParameters: null
+    backend: builtin # the storage option which stores already all other metadata. It could be the same Redis instance
```

Comment:
```suggestion
    backend: builtin # the storage option which stores already all other metadata. It could be the same Redis instance
```

this `builtin` is another TypeInstance or it's a keyword that we understand? maybe it will be good to unify that with Impl concept and add the `abstract: true` property?

Reply:
This is what I clarified during your review 😄 Please take a look.

While I like unification, I don't think `abstract: true` would be self-explanatory in this case 🤔